code/lib/core-server/src/utils/StoryIndexGenerator.deprecated.test.ts

```diff
@@ -1152,6 +1152,40 @@ describe('StoryIndexGenerator with deprecated indexer API', () => {
 
         expect(logger.warn).not.toHaveBeenCalled();
       });
+
+      it('DOES NOT throw when the same CSF file is indexed by both a deprecated and current indexer', async () => {
```

Comment:
You could have both the deprecated `storyIndexers` and the new `indexers` trying to index the same file, albeit it's a bit unlikely. In this scenario I've currently chosen that the new indexers takes precedence and the deprecated storyIndexers would be ignored for such a file (thus only a single index entry is generated), but I'm open to other suggestions, such as throwing an error instead.